ballerina/types.bal

```diff
@@ -13,50 +13,115 @@
+# managed Redis databases, where the cloud provider usually provides a connection URI.
+#
+# + uri - The connection URI for the Redis database
+@display {label: "Connection URI"}
+type ConnectionUri string;
+
 # Connection options for Redis client endpoint.
 #
-# + clientName - The clientName of the connection
-# + connectionPooling - Boolean value depending on whether the connection
-# pooling is enabled or not
-# + isClusterConnection - Whether to enable cluster connection or not
-# + ssl - Boolean value depending on whether SSL is enabled or not
-# + startTls - Boolean value depending on whether startTLS is enabled or not
-# + verifyPeer - Boolean value depending on whether peer verification is
-# enabled or not
-# + database - The database to be used with the connection
-# + connectionTimeout - The timeout value for the connection
+# + clientName - Name of the client
+# + database - Database index which the client should interact with. Not applicable for cluster connections
+# + connectionTimeout - Connection timeout in seconds
+@display {label: "Connection Options"}
 public type Options record {|
     @display {label: "Client Name"}
     string clientName?;
-    @display {label: "Connection Pooling Enabled"}
-    boolean connectionPooling = false;
-    @display {label: "Cluster Connection Enabled"}
-    boolean isClusterConnection = false;
-    @display {label: "SSL Enabled"}
-    boolean ssl = false;
-    @display {label: "STARTTLS Enabled"}
-    boolean startTls = false;
-    @display {label: "Peer Verification Enabled"}
-    boolean verifyPeer = false;
-    @display {label: "Database"}
-    int database = -1;
+    @display {label: "Database Index"}
+    int database = 0;
     @display {label: "Connection Timeout"}
-    int connectionTimeout = -1;
+    int connectionTimeout = 60;
+|};
+
+# Configurations for secure communication with the Redis server.
+#
+# + cert - Configurations associated with `crypto:TrustStore` or single certificate file that the client trusts  
+# + key - Configurations associated with `crypto:KeyStore` or combination of certificate and private key of the client  
+# + protocols - List of protocols used for the connection established to Redis Server, such as TLSv1.2, TLSv1.1, TLSv1.
+# + ciphers - List of ciphers to be used for SSL connections
+# + verifyMode - The SSL/TLS verification mode. This can be either NONE, CA, or FULL.
+# + startTls - Whether StartTLS is enabled
+@display {label: "Secure Socket Configurations"}
+public type SecureSocket record {|
+    @display {label: "Certificate"}
+    crypto:TrustStore|string cert;
+    @display {label: "Key"}
+    crypto:KeyStore|CertKey key?;
+    @display {label: "Protocols to be used for the connection to the Redis Server"}
+    string[] protocols?;
+    @display {label: "Ciphers"}
+    string[] ciphers?;
+    @display {label: "Peer Verification Mode"}
+    SslVerifyMode verifyMode = NONE;
```

Comment:
Shall we keep the "FULL" as the default mode? If we set it to 'NONE', even if the user configures everything else (like certificates, keys, ciphers, etc.), they still need to specifically set it to 'CA' or 'FULL' to enable validation.

Reply:
Yes that makes sense. Changed the default behaviour with https://github.com/ballerina-platform/module-ballerinax-redis/pull/218/commits/339c0b8c9a7a4261e4d363864b4f8416527e8dfe